api/google_api/routes/Mailer.js

```diff
@@ -2,41 +2,67 @@ const express = require('express');
 const router = express.Router();
 const { SceGoogleApiHandler } = require('../util/SceGoogleApiHandler');
 const { verification } = require('../email_templates/verification');
+const { blastEmail } = require('../email_templates/blastEmail');
 const {
   OK,
   BAD_REQUEST
 } = require('../../util/constants').STATUS_CODES;
 const { googleApiKeys } = require('../../config/config.json');
 const { USER } = googleApiKeys;
 
+// Routing post /sendVerificationEmail calls the sendEmail function
+// and sends the verification email with the verification email template
 router.post('/sendVerificationEmail', async (req, res) => {
   const scopes = ['https://mail.google.com/'];
   const pathToToken = __dirname + '/../../config/token.json';
-  const apiHandler = new SceGoogleApiHandler(
-    scopes, pathToToken);
+  const apiHandler = new SceGoogleApiHandler(scopes, pathToToken);
   const tokenJson = await apiHandler.checkIfTokenFileExists();
 
   if (tokenJson) {
     if (apiHandler.checkIfTokenIsExpired(tokenJson)) {
       apiHandler.refreshToken();
     }
   } else {
-    apiHandler.getNewToken(false);
+    apiHandler.getNewToken();
   }
 
-  await verification(
-    USER, req.body.recipientEmail, req.body.recipientName
+  await verification(USER, req.body.recipientEmail, req.body.recipientName)
+    .then((template) => {
+      apiHandler
+        .sendEmail(template)
+        .then((_) => {
+          res.sendStatus(OK);
+        });
+    })
+    .catch((_) => {
+      res.sendStatus(BAD_REQUEST);
+    });
+});
+
+// Routing post /sendBlastEmail calls the sendEmail function
+// and sends the blast email with the blast email template
+router.post('/sendBlastEmail', async (req, res) => {
+  const scopes = ['https://mail.google.com/'];
+  const pathToToken = __dirname + '/../../config/token.json';
+  const apiHandler = new SceGoogleApiHandler(scopes, pathToToken);
+
+  await blastEmail(
+    USER,
+    req.body.recipientEmail,
+    req.body.blastSubject,
+    req.body.blastContent
   )
     .then((template) => {
-      apiHandler.sendEmail(template)
-        .then(_ => {
+      apiHandler
+        .sendEmail(template)
+        .then((_) => {
           res.sendStatus(OK);
         })
-        .catch(_ => {
+        .catch((_) => {
```

Comment:
you can get rid of this inner `.catch` as the outer one will catch all errors

Reply:
Removing the inner `.catch` caused an error in running the api-test so we decided to keep it for now. If there are any solutions, we can figure it out later

Reply:
no problem